From cd1edee7b18368926250b31ef155a29455f6cf09 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E5=80=9E=E8=AF=9A?= <a591951062@gmail.com>
Date: Tue, 5 Sep 2023 07:23:10 +0000
Subject: [PATCH] Fix NullPointerException in TextView#canPasteAsPlainText

getPrimaryClipDescription may return null when permission is denied or in other abnormal circumstances.

Change-Id: If6b8cb935475fadced351ea7c1e585d99465b559
---
 core/java/android/widget/TextView.java | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/java/android/widget/TextView.java b/core/java/android/widget/TextView.java
index 450bb1e77ec8..81320a452e94 100644
--- a/core/java/android/widget/TextView.java
+++ b/core/java/android/widget/TextView.java
@@ -13355,6 +13355,9 @@ public class TextView extends View implements ViewTreeObserver.OnPreDrawListener
 
         final ClipDescription description =
                 getClipboardManagerForUser().getPrimaryClipDescription();
+        if (description == null) {
+            return false;
+        }
         final boolean isPlainType = description.hasMimeType(ClipDescription.MIMETYPE_TEXT_PLAIN);
         return (isPlainType && description.isStyledText())
                 || description.hasMimeType(ClipDescription.MIMETYPE_TEXT_HTML);
-- 
2.48.1

